core/src/main/resources/lib/layout/pageHeader.jelly

```diff
@@ -0,0 +1,137 @@
+Michelin, Romain Seguy, CloudBees, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:i="jelly:fmt" xmlns:x="jelly:xml">
+    <st:documentation>
+        Generates the page header, along with its associated dynamic behaviours.
+        This tag is used by l:layout and not expected to be used by anyone else,
+        but it's written as separate tag for better readability of code.
+
+        <st:attribute name="title" required="true">
+            Page title and title attribute for the logo
+        </st:attribute>
+
+        <st:attribute name="logoAlt" required="true">
+            Alt text for the logo
+        </st:attribute>
+
+        <st:attribute name="searchPlaceholder" required="true">
+            Placeholder text for the search input
+        </st:attribute>
+
+        <st:attribute name="searchHelpUrl" required="true">
+            Link value for the help icon on the search box
+        </st:attribute>
+
+        <st:attribute name="logout" required="true">
+            Text for the logout link
+        </st:attribute>
+    </st:documentation>
+
+    <header id="header" class="page-header">
+        <div class="page-header__brand">
+            <div class="logo">
+                <a id="jenkins-home-link" href="${rootURL}/">
+                    <img id="jenkins-head-icon" src="${imagesURL}/jenkins-header-logo-new.svg" alt="[${logoAlt}]" />
+                    <img id="jenkins-name-icon" src="${imagesURL}/title.png" alt="${title}" width="139" height="34" />
+                </a>
+            </div>
+
+            <a class="page-header__brand-link" href="${rootURL}/">
+                <img src="${imagesURL}/jenkins-header-logo-new.svg"
+                     alt="[${logoAlt}]"
+                     class="page-header__brand-image" />
+                <span class="page-header__brand-name">Jenkins</span>
```

Comment:
Is there a i18n property for the product name?

Reply:
No, Jenkins is Jenkins in all languages.